src/keboola/component/dao.py

```diff
@@ -435,16 +552,73 @@ def _filter_attributes_by_manifest_type(self, manifest_type: Literal["in", "out"
 
-        new_dict = self._raw_manifest.copy()
-        if supported_fields:
-            for attr in self._raw_manifest:
-                if attr not in supported_fields:
-                    new_dict.pop(attr, None)
-        return new_dict
+        if isinstance(self, TableDefinition):
+            supported_fields = self._manifest_attributes.get_attributes_by_stage(manifest_type, legacy_queue,
+                                                                                 native_types)
+            fields = {
+                'id': self.id,
+                'uri': self._uri,
+                'name': self.name,
+                'created': self._created,
+                'last_change_date': self._last_change_date,
+                'last_import_date': self._last_import_date,
+                'rows_count': self._rows_count,
+                'data_size_bytes': self._data_size_bytes,
+                'is_alias': self._is_alias,
+
+                'destination': self.destination,
+                'columns': self.columns if native_types else self.legacy_columns,
+                'incremental': self.incremental,
+                'primary_key': self.primary_key,
+                'write_always': self.write_always,
+                'delimiter': self.delimiter,
+                'enclosure': self.enclosure,
+                'metadata': self.table_metadata.get_table_metadata_for_manifest(),
+                'column_metadata': self.table_metadata.get_column_metadata_for_manifest(),
+                'manifest_type': manifest_type,
+                'has_header': self._has_header_in_file(),
+                'description': None,
+                'table_metadata': self.table_metadata.get_table_metadata_for_manifest(new_manifest=True),
+                'delete_where_column': self.delete_where_column,
+                'delete_where_values': self.delete_where_values,
+                'delete_where_operator': self.delete_where_operator,
+                'schema': [col.to_dict() for col in self.schema] if self.schema else []
+            }
+
+            new_dict = fields.copy()
+
+            if supported_fields:
+                for attr in fields:
+                    if attr not in supported_fields:
+                        new_dict.pop(attr, None)
+            return new_dict
 
-    def get_manifest_dictionary(self, manifest_type: Optional[str] = None, legacy_queue=False) -> dict:
+        else:
+            return {
+                'id': self.id,
+                'created': self.created.strftime('%Y-%m-%dT%H:%M:%S%z') if self.created else None,
+                'is_public': self.is_public,
+                'is_encrypted': self.is_encrypted,
+                'name': self.name,
+                'size_bytes': self.size_bytes,
+                'tags': self.tags,
+                'notify': self.notify,
+                'max_age_days': self.max_age_days,
+                'is_permanent': self.is_permanent,
+            }
+
+    def _has_header_in_file(self):
```

Comment:
@kudj tady jsem myslel, ze jsme si odsouhlasili, ze `has_headers` bude property TableDefinition a pujde nastavit userem. Ta magic logika se `self.columns` uz neplati (jen pro stary manifesty, takze se ti ta metoda porad bude hodit).

Reply:
@kudj co tenhle komentar? porad se to nechova jako predtim (tj set columns nenastavi sloupce, ale tise to vyhodi warning kdyz tam je schema, nebo to prida jenom nektery